### PR TITLE
feat:  Support more `keyof` for `IndexedAccessType`s

### PR DESCRIPTION
**Pull request by sunrabbit123**
**Description:**

```ts
function f90<T extends S2, K extends keyof S2>(x1: S2[keyof S2], x2: T[keyof S2], x3: S2[K]) {
    x1 = x2;
    x1 = x3;
    x2 = x1;
    x2 = x3;
    x3 = x1;
    x3 = x2;
    x1.length;
    x2.length;
    x3.length;
}
```

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: crates/stc_ts_file_analyzer/tests/tsc/types/nonPrimitive/.1.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// @strict: true
```
Comment (kdy1) [inline crates/stc_ts_file_analyzer/tests/tsc/types/nonPrimitive/.1.ts:1]: Seems like this test is ignored?

Comment (kdy1) [reply crates/stc_ts_file_analyzer/tests/tsc/types/nonPrimitive/.1.ts:1]: Please unignore this, or delete `1.tsc-errors.json`

**Review by kdy1 — CHANGES_REQUESTED** [review]

File: crates/stc_ts_file_analyzer/tests/tsc/types/nonPrimitive/.1.ts (hunk 1 above)
Comment (sunrabbit123) [reply crates/stc_ts_file_analyzer/tests/tsc/types/nonPrimitive/.1.ts:1]: sir, this ignored test file mean decrease matched error

Comment (sunrabbit123) [reply crates/stc_ts_file_analyzer/tests/tsc/types/nonPrimitive/.1.ts:1]: I will remove errors.json, thanks

**Review by kdy1 — APPROVED** [review]
Thank you!

Can you rebase, please?

**Review by kdy1 — APPROVED** [review]

**Pull request merged**